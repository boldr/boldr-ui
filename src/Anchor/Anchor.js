/* @flow */
import React, { Component } from 'react';
import classnames from 'classnames';
import Link from 'react-router-dom/Link';

type Props = {
  href: string,
  className: ?string,
  label: ?string,
  children: ?any,
  onClick: ?() => void,
};

const Anchor = (props: Props) => {
  const { href, className, label, onClick, children } = props;
  return (
    <Link to={href} className="boldrui-link" onClick={onClick} title={label} {...props}>
      {label || children}
    </Link>
  );
};

export default Anchor;
