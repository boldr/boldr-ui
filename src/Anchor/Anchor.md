# Anchor

Nothing more exciting than a wrapped React-Router Link component with a class name attached to it.

### Usage
  <Anchor
    href="/some-page-over-there"
    className="anchorage"
    label="AK"
  />

### Props

| Name            | Type                     | Default      | Required   | Description |
|-----------------|--------------------------|--------------|------------|-------------|
| children        | ReactChildren            |              |   ----     |   ----      |
| href            | string                   |              |   true     |  URL you are linking to.          |
| className       | string                   |    ----      |   ----     |  css class name         |
| label           | string                   |    ----      |   ----     |  text for the link. You may use label or children, but not both.        |
