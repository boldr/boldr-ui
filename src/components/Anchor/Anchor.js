/* @flow */
import React, { Component } from 'react';
import classnames from 'classnames';
import NavLink from 'react-router-dom/NavLink';

type Props = {
  href: string,
  className: string,
  label: string,
  children: any,
  onClick: () => void,
  icon: any,
};

const Anchor = (props: Props) => {
  const { href, className, label, onClick, icon } = props;
  return (
    <NavLink
      to={href}
      className="boldrui-link"
      onClick={onClick}
      title={label}
      {...props}
    >
      {props.icon ? <i className={classnames('fa', props.icon)} /> : null}
      {label}
    </NavLink>
  );
};

export default Anchor;
