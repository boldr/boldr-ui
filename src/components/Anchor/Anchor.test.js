import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Anchor from './Anchor';

describe('<Anchor />', () => {
  const wrapper = shallow(<Anchor href="/" title="Home" onClick={() => {}} />);
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
