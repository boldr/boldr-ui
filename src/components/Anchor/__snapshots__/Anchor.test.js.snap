// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> renders snapshot 1`] = `
<NavLink
  activeClassName="active"
  className="boldrui-link"
  href="/"
  onClick={[Function]}
  title="Home"
  to="/"
/>
`;
